### PR TITLE
Feature/ens

### PR DESCRIPTION
**Pull request by nishil-edx**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,49 +1 @@
...
-Note that the `prepublishOnly` script will be automatically called while publishing. It will check lint, clean `dist/lib` folders, and build fresh distribution files before executing `npm publish`.
-
-```bash
-npm publish
-```
-
-## License
-
-MIT
+This branch mainly targeted on ENS interactions
```
Comment (ex-rohan-edx) [inline README.md]: Revert back to old READ ME file

File: src/classes/EdexaClient.ts [hunk 2]
```diff
@@ -144,33 +144,50 @@ export class EdexaClient {
...
+  async resolveAddr(input: string) {
     try {
       // Check if the input is a valid Ethereum address
       if (ethers.utils.isAddress(input)) {
-        return input
+       
+        const provider = new ethers.providers.JsonRpcProvider(RPC_URL)
+        const reverseName = `${input.slice(2)}.addr.reverse`;
+        const node = ethers.utils.namehash(reverseName);
+        const resolverContract_ = new ethers.Contract('0xEF1db68FaDfdD398886EE73Cbc34918Be51Ae304', publicResolver.abi, provider);
```
Comment (ex-rohan-edx) [inline src/classes/EdexaClient.ts]: Move static contract address into a constants file

File: src/classes/EdexaClient.ts [hunk 3]
```diff
@@ -144,33 +144,50 @@ export class EdexaClient {
...
-
-        if (detailsObject) {
-          if (
-            detailsObject.resolver ==
-            '0x0000000000000000000000000000000000000000'
-          )
-            throw new Error(`ENS Not Registered for ${input}`)
-          else return detailsObject.resolver
+        const node = ethers.utils.namehash(input);
+        const resolverContract_ = new ethers.Contract('0xEF1db68FaDfdD398886EE73Cbc34918Be51Ae304', publicResolver.abi, provider);
```
Comment (ex-rohan-edx) [inline src/classes/EdexaClient.ts]: Move static contract address into a constants file

File: src/classes/EdexaClient.ts [hunk 4]
```diff
@@ -144,33 +144,50 @@ export class EdexaClient {
...
+        
+      }
+    } catch (error) {
+      throw new Error(`Error: ${error.message}`)
+    }
+  }
+  async resolveName(input: string) {                  //name string input
+    try {
+      // Check if the input is a valid Ethereum address
+      if (input.endsWith(".edx") && input.split(".")[0].length >= 4) {
```
Comment (ex-rohan-edx) [inline src/classes/EdexaClient.ts]: Move static string checking var to a constants file

File: testENSresolve.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+async function resolveName(input : string) {
+    
+    let edexaclient = new EdexaClient();
+    let add = await edexaclient.resolveName(input);
+    console.log("address of",input, " is:",add)
+}
+
+
+
+resolveAddress("0x44aDbAcC4B0baD81dBc6D24b9512da3806c5aCCD")
```
Comment (ex-rohan-edx) [inline testENSresolve.ts:25]: Add more test cases to check the functionality for example add negative test cases too

File: tsconfig.json [hunk 6]
```diff
@@ -18,6 +18,6 @@
     "esModuleInterop": true,
     "declaration": true
   },
-  "include": ["./src/**/*"],
+  "include": ["./src/**/*", "testENSresolve.ts"],
```
Comment (ex-rohan-edx) [inline tsconfig.json]: Reset back to original file

**Review by ex-rohan-edx — CHANGES_REQUESTED** [review]

**Comment by nishil-edx** [comment]
updated

**Review by ex-rohan-edx — CHANGES_REQUESTED** [review]
**Function Comments:**
     - Add comments to each function describing:
     - The function's purpose.
     - The types of arguments it accepts.
     - What the function returns.

**Test Case Improvement:**
   - Specify the expected results in your test cases to clarify what the test is verifying.
   - Use descriptive names for your test cases to indicate the aspect of the function being tested.
